packages/medusa/src/services/new-totals.ts

```diff
@@ -112,7 +112,7 @@ export default class NewTotalsService extends TransactionBaseService {
           lineItemsTaxLinesMap[item.id] = item.tax_lines ?? []
         })
       } else {
-        if (items.some((item) => !item.variant)) {
+        if (items.some((item) => item.variant_id && !item.variant)) {
```

Comment:
thought: We have guards against tax calculation on custom items [here](https://github.com/medusajs/medusa/blob/5250a231f7af4d8187295707c605fc601ef903b1/packages/medusa/src/services/tax-provider.ts#L266), so we could change the condition to allow custom items to pass through here.

Same goes for the other change in the PR.

Let me know what you think

Reply:
Yes you are right, in that case I think we have to remove the guards in that service. The one that throw here for the variant and the one that throws l203. wdyt?

Reply:
Sounds reasonable to me. Can you cc the original contributor of the check, so we can be sure?

Reply:
I believe the check should be added [here](https://github.com/medusajs/medusa/blob/5250a231f7af4d8187295707c605fc601ef903b1/packages/medusa/src/services/tax-provider.ts#L266). 

If `variant_id` is present but not the `variant`, we should throw.

Reply:
The original check in the totals service where checking if it was explicitly undefined and then throw a must be joined error

Reply:
@srindom as the original author of the check in the totals service. we said that if we have an order and tax lines are undefined then we throw. In the case of custom line items, we might not have the tax lines. wdyt?

Reply:
What check are you referring to here @adrien2p? Custom line items should not hold tax lines.

Reply:
@olivermrbl yes I believe it should be enough to have it in the tax-provider

Reply:
I am referring to this one https://github.com/medusajs/medusa/blob/master/packages/medusa/src/services/totals.ts#L842 which is the original check. It stands that if it is an order and the item does not have a tax lines prop then it throws

Reply:
@olivermrbl I believe the changes I made to the check in the `getLineItemTotals_` should be good. We don't want to throw in case there is not tax lines and not variant_id. wdyt?